### PR TITLE
Fixed TRIM/LTRIM/RTRIM/BTRIM on null 2nd arg

### PR DESCRIPTION
**Pull request by matriv**
Return `NULL` not only if the input string to trim is `NULL`, but also if the `trimmingText` argument of the functions is `NULL`, thus complying with PostgreSQL behaviour.

Fixes: #15527

### CODE REVIEW

File: server/src/test/java/io/crate/expression/scalar/string/TrimFunctionTest.java [hunk 1]
```diff
@@ -148,10 +148,21 @@ public void testEvaluateTrimWithoutCharsToTrim() {
     @Test
     public void testEvaluateNullInputOptimisedTrim() {
         assertEvaluateNull("trim(name)", Literal.of(DataTypes.STRING, null));
+        assertEvaluateNull("trim('foo' FROM name)", Literal.of(DataTypes.STRING, null));
         assertEvaluateNull("ltrim(name)", Literal.of(DataTypes.STRING, null));
         assertEvaluateNull("rtrim(name)", Literal.of(DataTypes.STRING, null));
+        assertEvaluateNull("btrim(name)", Literal.of(DataTypes.STRING, null));
         assertEvaluateNull("ltrim(name, null)", Literal.of(DataTypes.STRING, null));
         assertEvaluateNull("rtrim(name, null)", Literal.of(DataTypes.STRING, null));
+        assertEvaluateNull("btrim(name, null)", Literal.of(DataTypes.STRING, null));
```
Comment (jeeminso) [inline server/src/test/java/io/crate/expression/scalar/string/TrimFunctionTest.java:157]: Are these 3 new asserts just to cover all variations? They seem unrelated to the bug fix.

**Review by jeeminso — APPROVED** [review]
LGTM, just one question, Thanks!

File: server/src/test/java/io/crate/expression/scalar/string/TrimFunctionTest.java (hunk 1 above)
Comment (matriv) [reply server/src/test/java/io/crate/expression/scalar/string/TrimFunctionTest.java:157]: yep!

Comment (matriv) [reply server/src/test/java/io/crate/expression/scalar/string/TrimFunctionTest.java]: Please take on more look (squashed commit already)
I change the 2nd arg on this method from `null` to `foo`, to separate visibly that the `null` on the 2nd arg is tested in the new method. thx!

Comment (jeeminso) [reply server/src/test/java/io/crate/expression/scalar/string/TrimFunctionTest.java]: Looks good 👍
